### PR TITLE
DCOS-43338: set correct reference branch for dcos repo

### PR DESCRIPTION
**Pull request by juliangieseke**
.dcosrc file referenced `dcos/dcos` branches, not `dcos/dcos-ui` ones - its the branch our `dcos/dcos` PRs are against.

changed setup slightly to pass in base branch from semantic-release, that way we only have to update one file.

## Testing

well… once this is merged, building latest PR on `release/1.12` should work 🤷‍♂️

### CODE REVIEW

**Comment by juliangieseke** [comment]
`.releaserc` file also needs update.

**Comment by nLight** [comment]
Is it documented anywhere?

**Comment by juliangieseke** [comment]
nope, give me a second to streamline everything - I'll add documentation to Wiki once this PR is merged.

**Comment by nLight** [comment]
shall we consolidate those https://github.com/dcos/dcos-ui/pull/3348 ?

**Comment by juliangieseke** [comment]
@nLight no, different issue/topic I'd say.

please revisit, I'll port the changes to master & 1.11 and create documentation once this is merged.

File: scripts/ci/upload-release [hunk 1]
```diff
@@ -28,14 +28,12 @@ SCRIPT_PATH="$(cd $(dirname "$0")/$(dirname "$(readlink "$0")") && pwd)"
...
 
-# get correct DC/OS target (=base branch!) from .dcosrc file
-PKG_TARGET=$(python -c "import sys, json; \
-  dcosrc = json.load(open('$PROJECT_ROOT/.dcosrc')); \
-  sys.stdout.write(str(dcosrc['base']));")
-
 # next release version (passed in by semantic-release)
 PKG_VERSION="${1}"
 
+# get correct DC/OS target (=base branch!) (passed in by semantic-release)
```
Comment (nLight) [inline scripts/ci/upload-release]: Why can't we use base branch? How would it look like on master?

File: .releaserc [hunk 2]
```diff
@@ -1,12 +1,12 @@
 {
-  "branch": "master",
-  "tagFormat": "master+v${version}",
+  "branch": "release/1.12",
```
Comment (nLight) [inline .releaserc]: I wonder if we could have this as ENV vars in Jenkinsfile it's just tooooooooooo many places to look at.

Comment (juliangieseke) [reply .releaserc]: its a JSON file :(

Comment (juliangieseke) [reply .releaserc]: Also I try to consolidate all information in that very file to have less places to look at 👍

Comment (nLight) [reply .releaserc]: We could switch to a `js` file https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/configuration.md#configuration-file

File: scripts/ci/upload-release (hunk 1 above)
Comment (juliangieseke) [reply scripts/ci/upload-release]: our branch is named `release/1.12` while dcos branch is named just `1.12` - I had a fance regex before, but that was too complicated in the end…

File: .releaserc (hunk 2 above)
Comment (juliangieseke) [reply .releaserc]: uh, nice! but lets not do it in this PR please.

Comment (nLight) [reply .releaserc]: deal

File: scripts/ci/upload-release (hunk 1 above)
Comment (nLight) [reply scripts/ci/upload-release]: but we could read this data from `.dcosrc` can't we?

Comment (juliangieseke) [reply scripts/ci/upload-release]: My idea was to get rid of this file and have all the information in `.releaserc` 🤔

Comment (juliangieseke) [reply scripts/ci/upload-release]: @nLight anything else preventing you from ✅?

Comment (nLight) [reply scripts/ci/upload-release]: Did we capture all the improvements in a JIRA?

Comment (juliangieseke) [reply scripts/ci/upload-release]: nope, shall I create one?

Comment (nLight) [reply scripts/ci/upload-release]: would appreciate it

Comment (juliangieseke) [reply scripts/ci/upload-release]: DCOS-43338

**Comment by juliangieseke** [comment]
added JIRA issue - @nLight @GeorgiSTodorov 🙇

File: scripts/ci/upload-release (hunk 1 above)
Comment (nLight) [reply scripts/ci/upload-release]: I'm sorry I mean a JIRA to further improve what we've discussed here. Like consolidating variables for a release number etc.

**Pull request merged**